pallets/nfts/src/lib.rs

```diff
@@ -465,21 +498,31 @@ pub mod pallet {
 		/// a `delegate`.
 		TransferApproved {
 			collection: T::CollectionId,
-			item: T::ItemId,
+			item: Option<T::ItemId>,
 			owner: T::AccountId,
 			delegate: T::AccountId,
 			deadline: Option<BlockNumberFor<T>>,
 		},
-		/// An approval for a `delegate` account to transfer the `item` of an item
-		/// `collection` was cancelled by its `owner`.
+		/// An approval for a `delegate` account to transfer a specific `item` in a `collection` or
+		/// all collection items owned by the `owner` has been cancelled by the owner.
 		ApprovalCancelled {
 			collection: T::CollectionId,
-			item: T::ItemId,
+			item: Option<T::ItemId>,
```

Comment:
Note: Tested in: 

- cancel_approval_works: https://github.com/r0gue-io/pop-node/pull/387/files#r1881283901